File: frontend/src/services/api/tasks.ts
import { fetchClient } from './client';
import type { OrderTask, PaginatedResponse } from './types';

export interface TaskFilters {
  search?: string;
  status?: string;
  department?: string;
  stage?: string;
  order_id?: string;
}

export const tasksApi = {
  async getTasks(filters: TaskFilters = {}, page: number = 1, pageSize: number = 20): Promise<PaginatedResponse<OrderTask>> {
    const params = new URLSearchParams();
    
    params.append('page', page.toString());
    params.append('page_size', pageSize.toString());
    
    if (filters.search) params.append('search', filters.search);
    if (filters.status) params.append('status', filters.status);
    if (filters.department) params.append('department', filters.department);
    if (filters.stage) params.append('stage', filters.stage);
    if (filters.order_id) params.append('order_id', filters.order_id);
    
    return fetchClient<PaginatedResponse<OrderTask>>(`/api/tasks?${params.toString()}`);
  }
};

File: backend/app/api/routes/tasks.py
from typing import Optional
from fastapi import APIRouter, Depends, Query
from app.api.dependencies import get_oms_service
from app.oms.services.oms_service import OMSService
from app.oms.schemas.oms import OrderTaskSchema
from app.oms.schemas.api import PaginatedResponse
from app.oms.contracts.queries import TaskQuery

router = APIRouter(prefix="/api/tasks", tags=["Tasks"])

@router.get("", response_model=PaginatedResponse[OrderTaskSchema])
def list_tasks(
    search: Optional[str] = Query(None, description="Search term for tasks"),
    status: Optional[str] = Query(None, description="Exact match on task status"),
    department: Optional[str] = Query(None, description="Exact match on department"),
    stage: Optional[str] = Query(None, description="Exact match on stage"),
    order_id: Optional[str] = Query(None, description="Filter by order ID"),
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    service: OMSService = Depends(get_oms_service)
):
    """Retrieve a paginated list of OMS tasks with optional filtering."""
    q = TaskQuery(
        search=search,
        status=status,
        department=department,
        stage=stage,
        order_id=order_id,
        page=page,
        page_size=page_size
    )
    return service.list_tasks(q)
